packages/victory-pie/src/helper-methods.ts

```diff
@@ -59,6 +59,26 @@ const getSlices = (props, data) => {
   return layoutFunction(data);
 };
 
+// sorts data by the categories prop. if all of the data keys aren't included in categories,
+// any remaining data will be appended to the data array.
+// if extraneous categories are included in the categories prop, the will be ignored and have no effect
+// on the rendered component.
+const sortDataByCategories = (props, data) => {
+  if (Array.isArray(props?.categories?.x)) {
+    const sorted: string[] = [];
+    props.categories.x.forEach((category) => {
+      const idx = data.findIndex(({ x }) => x === category);
+      if (idx >= 0) {
+        const datum = data.splice(idx, 1)[0];
+        sorted.push(datum);
+      }
+    });
+    return [...sorted, ...data];
+  }
+
+  return data;
+};
```

Comment:
No mention of categories in `VictoryBar` or it's helper methods, but I'm digging around in some other components to see what I can find. Chart has them and is definitely respecting them, but I'm still looking for the logic that makes it all work.

Reply:
After digging in to `VictoryChart`, it ends up retrieving and parsing categories through `getCalculatedProps` in its `helper-methods`, which calls `getStringsFromChildren` from the `Wrapper` util, then passes the result of that call ( `{x: string[], y: string[] }`) along with `props` and `childComponents` to `Wrapper.getCategories` to retrieve an object with the `{x: string[], y: string[] }` shape. That object is then passed to `createStringMap` in the helper methods for `VictoryChart`, which returns an object of keys mapped to order (`{ [string]: [number] }`), and then sends that object and other props down to the `VictoryChart`'s child components. It doesn't seem like `PieChart` will have any child components and the code I've added here seems easier to follow to me, but I could use `Wrapper.getCategories` here if you think that makes more sense (I have updated the code in this PR to handle `props.categories` as `string[]`).


I did some more looking into `VictoryBar` too, and it does support categories, I'd been looking for a direct reference in it's code rather than into utility methods. I didn't get as deep into the call chain as with `VictoryChart`, but there are some visual bugs when using scale and categories together, which might be worth opening another issue to address:

```javascript
          <VictoryBar
            theme={VictoryTheme.clean}
            style={{
              data: { fill: VictoryTheme.clean.palette?.colors?.red },
            }}
            scale={{ y: "log", x: "linear" }}
            horizontal
            categories={{
              x: ["a", "b", "c", "d", "e", "f", "g", "h"],
              y: ["i", "j", "k", "l", "m", "n", "o", "p"],
            }}
            data={[
              { x: 1, y: 0.1 },
              { x: 2, y: 1 },
              { x: 3, y: 10 },
              { x: 4, y: 0 },
              { x: 5, y: 0.1 },
              { x: 6, y: 1 },
              { x: 7, y: 10 },
              { x: 8, y: 100 },
            ]}
          />
```
<img width="310" alt="image" src="https://github.com/user-attachments/assets/f0051107-ba75-4cd7-bff1-07a1e3b4d030">

Without scale, both x and y categories work fine:

```javascript
          <VictoryBar
            theme={VictoryTheme.clean}
            style={{
              data: { fill: VictoryTheme.clean.palette?.colors?.red },
            }}
            // scale={{ y: "log", x: "linear" }}
            horizontal
            categories={{
              x: ["a", "b", "c", "d", "e", "f", "g", "h"],
              y: ["i", "j", "k", "l", "m", "n", "o", "p"],
            }}
            data={[
              { x: 1, y: 0.1 },
              { x: 2, y: 1 },
              { x: 3, y: 10 },
              { x: 4, y: 0 },
              { x: 5, y: 0.1 },
              { x: 6, y: 1 },
              { x: 7, y: 10 },
              { x: 8, y: 100 },
            ]}
          />
```
<img width="486" alt="image" src="https://github.com/user-attachments/assets/fb1616f6-095d-4869-9238-4ee8b3c52e0d">

Reply:
There's also this build issue I'll need to look into.